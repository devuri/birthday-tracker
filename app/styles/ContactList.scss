contactList {
  align-items: center;
  background-color: #09C6AB;
  flex: 1;
  justify-content: center;
  padding-bottom: 60px;
}

searchBar {
  background-color: #EAE8E8;
  border-radius: 10px;
  margin: 80px 0 10px 0;
  width: 300px;
}

searchInput {
  color: #02556D;
  height: 40px;
  text-align: center;
}
