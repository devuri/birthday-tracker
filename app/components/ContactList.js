import React, { Component } from 'react';
import styles from '../styles/main';
import { ListView, Text } from 'react-native';

const contacts = [
  {firstName: "Kirsten", lastName: "Swanson", birthday: "ðŸŽˆ February 16th"},
  {firstName: "Mitch", lastName: "Davis", birthday: "ðŸŽˆ June 20th"}
];

export default class ContactList extends Component {
  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      dataSource: ds.cloneWithRows(contacts)
    };
  }

  render() {
    return (
      <ListView
        style={styles.contactList}
        dataSource={this.state.dataSource}
        renderRow={(contacts) => <Text>{contacts.firstName}{contacts.lastName}</Text>}
      />
    );
  }
}
