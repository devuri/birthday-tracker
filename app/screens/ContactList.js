import React, { Component } from 'react';
import { TextInput, View, AsyncStorage, ListView, Text, TouchableHighlight } from 'react-native';
import SearchContacts from '../components/SearchContacts';
import Button from '../components/Button';
import styles from '../styles/main';

const ContactList = React.createClass({
  getInitialState () {
    var contactInfo = new ListView.DataSource({rowHasChanged: (row1, row2) => row1 !== row2});
    var contacts = [];
    this.getStorage().then((contacts) => this.setItems(contacts));
    return {
      contacts,
      contactInfo: contactInfo.cloneWithRows(contacts),
      firstName: '',
      lastName: ''
    };
  },

  updateFirstName(e) {
    this.setState({firstName: e.nativeEvent.text});
  },

  updateLastName(e) {
    this.setState({lastName: e.nativeEvent.text});
  },

  onPress () {
    if (this.state.firstName) {
      this.state.contacts.push(this.state.firstName);
      this.setItems(this.state.contacts);
      this.setState({firstName: ''});
      this.updateStorage();
    }
    if (this.state.lastName) {
      this.state.contacts.push(this.state.lastName);
      this.setItems(this.state.contacts);
      this.setState({lastName: ''});
      this.updateStorage();
    }
  },

  getStorage () {
    return AsyncStorage
      .getItem('contacts')
      .then((contacts) => JSON.parse(contacts));
  },

  updateStorage () {
    return AsyncStorage.setItem('contacts', JSON.stringify(this.state.contacts));
  },

  setItems (contacts) {
    var contactInfo = this.state.contactInfo.cloneWithRows(contacts);
    this.setState({contacts, contactInfo});
  },

  render () {
    return (
      <View style={styles.contactList}>
        <SearchContacts />
        <ListView
          dataSource={this.state.contactInfo}
          renderRow={(rowData) => <Text>{rowData}</Text>}
        />
        <View style={styles.firstNameInput}>
          <TextInput
            style={styles.inputFields}
            placeholder="First Name"
            onChange={this.updateFirstName}
            value={this.state.firstName}
          />
        </View>
        <View style={styles.lastNameInput}>
          <TextInput
            style={styles.inputFields}
            placeholder="Last Name"
            onChange={this.updateLastName}
            value={this.state.lastName}
          />
        </View>
        <Button onPress={this.onPress} title="Save" />
      </View>
    );
  }
});

export default ContactList;
