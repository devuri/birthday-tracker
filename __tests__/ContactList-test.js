import 'react-native';
import React from 'react';
import ContactList from '../app/screens/ContactList.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ContactList />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
