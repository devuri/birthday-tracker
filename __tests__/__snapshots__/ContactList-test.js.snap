exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#09C6AB",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 60,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#EAE8E8",
        "borderRadius": 10,
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 80,
        "width": 300,
      }
    }>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChange={[Function]}
      placeholder="Search contacts"
      style={
        Object {
          "color": "#02556D",
          "height": 40,
          "textAlign": "center",
        }
      } />
  </View>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": "unknown",
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []} />
</View>
`;
